qucs/qucs/qucs.cpp

```diff
@@ -1436,7 +1436,17 @@ bool QucsApp::saveFile(QucsDoc *Doc)
     return saveAs();
 
   int Result = Doc->save();
-  if(Result < 0)  return false;
+  if(Result < 0)
+  {//The file was not saved.
+      if (Result == -2)
+      {//Save the document in another location
+          saveAs();
```

Comment:
this should be `return SaveAs()` since that function already takes care of `updatePortNumber()` and `slotUpdateTreeview()` which are done below here. Besides, currently if you cancel the "Save As" the functions continues and returns `true` and clears the diskette icon, as for a successful save.